pkg/provider/azure/data/vmsize.go

```diff
@@ -0,0 +1,57 @@
+package data
+
+import (
+	"context"
+	"os"
+	"slices"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v6"
+	"github.com/redhat-developer/mapt/pkg/util"
+)
+
+func IsVMSizeOfferedByLocation(vmSize, location string) (bool, error) {
+	offerings, err := FilterVMSizeOfferedByLocation([]string{vmSize}, location)
+	return len(offerings) == 1, err
+}
+
+// Get InstanceTypes offerings on current location
+func FilterVMSizeOfferedByLocation(vmSizes []string, location string) ([]string, error) {
+	// Create a new Azure credential (uses environment variables or managed identity)
```

Comment:
might be needed to use this here to convert the pulumi env variables starting with `AZURE_` to ones starting with `ARM_` since the azure sdk expects it to start with `ARM_` 

https://github.com/redhat-developer/mapt/blob/eaf24bec441c3f8751e1e2b436730b005bf29703/pkg/provider/azure/module/identity/azidentity.go#L19